src/daft-parquet/src/metadata.rs

```diff
@@ -213,73 +214,104 @@ fn apply_field_ids_to_parquet_file_metadata(
     })
 }
 
+pub(crate) fn validate_footer_magic(uri: &str, buffer: &[u8]) -> super::Result<()> {
+    const PARQUET_MAGIC: [u8; 4] = [b'P', b'A', b'R', b'1'];
+
+    if buffer[buffer.len() - 4..] != PARQUET_MAGIC {
```

Comment:
I believe some of the errors come from here for invalid files with less than 4 characters